src/utils.ts

```diff
@@ -823,3 +831,247 @@ export const messageSetPagination = <StreamChatGenerics extends ExtendableGeneri
+  channel: Channel<StreamChatGenerics>,
+) => {
+  if (!channel) return false;
+
+  const member = channel.state.membership;
+
+  return !!member?.pinned_at;
+};
+
+export const isChannelArchived = <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>(
+  channel: Channel<StreamChatGenerics>,
+) => {
+  if (!channel) return false;
+
+  const member = channel.state.membership;
+
+  return !!member?.archived_at;
+};
+
+export const shouldConsiderArchivedChannels = <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>(
+  filters: ChannelFilters<StreamChatGenerics>,
+) => {
+  if (!filters) return false;
+
+  return typeof filters.archived === 'boolean';
+};
+
+export const extractSortValue = <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>({
+  atIndex,
+  sort,
+  targetKey,
+}: {
+  atIndex: number;
+  targetKey: keyof ChannelSortBase<StreamChatGenerics>;
+  sort?: ChannelSort<StreamChatGenerics>;
+}) => {
+  if (!sort) return null;
+  let option: null | ChannelSortBase<StreamChatGenerics> = null;
+
+  if (Array.isArray(sort)) {
+    option = sort[atIndex] ?? null;
+  } else {
+    let index = 0;
+    for (const key in sort) {
+      if (index !== atIndex) {
+        index++;
+        continue;
+      }
+
+      if (key !== targetKey) {
+        return null;
+      }
+
+      option = sort;
+
+      break;
+    }
+  }
+
+  return option?.[targetKey] ?? null;
+};
+
+/**
+ * Returns true only if `{ pinned_at: -1 }` or `{ pinned_at: 1 }` option is first within the `sort` array.
+ */
+export const shouldConsiderPinnedChannels = <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>(
+  sort: ChannelSort<StreamChatGenerics>,
+) => {
+  const value = extractSortValue({
+    atIndex: 0,
+    sort,
+    targetKey: 'pinned_at',
+  });
+
+  if (typeof value !== 'number') return false;
+
+  return Math.abs(value) === 1;
+};
+
+export function findPinnedAtSortOrder<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>({
+  sort,
+}: {
+  sort: ChannelSort<StreamChatGenerics>;
+}) {
+  if (!sort) return null;
+
+  if (Array.isArray(sort)) {
+    const [option] = sort;
+
+    if (!option?.pinned_at) return null;
+
+    return option.pinned_at;
+  } else {
+    if (!sort.pinned_at) return null;
+
+    return sort.pinned_at;
+  }
+}
+
+export function findLastPinnedChannelIndex<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>({
+  channels,
+}: {
+  channels: Channel<StreamChatGenerics>[];
+}) {
+  let lastPinnedChannelIndex: number | null = null;
+
+  for (const channel of channels) {
+    if (!isChannelPinned(channel)) break;
+
+    if (typeof lastPinnedChannelIndex === 'number') {
+      lastPinnedChannelIndex++;
+    } else {
+      lastPinnedChannelIndex = 0;
+    }
+  }
+
+  return lastPinnedChannelIndex;
+}
+
+export const moveChannelUpwards = <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>({
```

Comment:
I would change the name as what the function does it moves the channel to determined index, but it does not need to be upwards as that is UI conotation.

```suggestion
export const changeChannelPosition = <StreamChatGenerics extends ExtendableGenerics = DefaultGenerics>({
```

Reply:
Yep - this method does not have the capability to move a `channel` any other direction than upwards (or at the very least make it stay in place). I do agree with what @MartinCupela is saying though, perhaps we need something better here as indeed `upwards` is a bit vague as a term to use for a list specifically.

Maybe something like `shiftChannelForward`, to be inline with JS lingo ?

Reply:
Proposal:

- `raiseChannelPriority`
- `promoteChannel`

Reply:
`promoteChannel` does actually sound pretty good, +1 from me. :)

Reply:
Done ! Thanks guys !